viadot/tasks/prefect_conformed.py

```diff
@@ -0,0 +1,147 @@
+import prefect
+from datetime import datetime, timedelta
+
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+from prefect.utilities import logging
+
+logger = logging.get_logger()
+
+class ReRunFailedFlow(Task):
+    def __init__(
+        self,
+        flow_name: str = None,
+        *args,
+        **kwargs,
+    ):
+
+        self.flow_name = flow_name
+
+        super().__init__(
+            name="prefect_extract_details",
+            *args,
+            **kwargs,
+        )
+
+    def __call__(self):
+        """Extract time from Prefect Flow run"""
+        super().__call__(self)
+    
+    def check_if_scheduled_run(self, created_by_user_id: str = None) -> bool:
+        """
+        Check if run was scheduled or started by user.
+
+        Args:
+            created_by_user_id (str, optional): User_id in prefect. Defaults to None.
+
+        Returns:
+            bool: True if flow run was started automatically. False if Flow was started by user.
+        """
+
+        if created_by_user_id == "09720c91-a99c-4f72-b7b5-3c061c83408b" :
+            return True
+        else:
+            return False
+
+        
+    def get_formatted_date(self, time_unclean: str = None):
+        """
+        Format date from "2022-03-08T01:00:00+00:00" to date or time.
+
+        Args:
+            time_unclean (str, optional): Time in datetime format obtained from Prefect. Defaults to None.
+
+        Returns:
+            List of two paramaters: [datetime.date, datetime.time]
+        """
+        date_format_str = "%Y-%m-%d %H:%M:%S"
+
+        time_extracted = time_unclean.split("T")[1]
+        time_clean_str = time_extracted.split(".")[0]
+        date_extracted = time_unclean.split("T")[0]
+
+        datetime_clean = datetime.strptime("{} {}".format(date_extracted, time_clean_str[:8]), date_format_str)
+
+        return datetime_clean 
```

Comment:
Maybe a good approach will be to move this method outside the class. This function can be combined with the existing one (in prefect.py). function in `get_formatted_date()` `prefect.py` have 2 options `date` and `time` - you can add a third code execution option.